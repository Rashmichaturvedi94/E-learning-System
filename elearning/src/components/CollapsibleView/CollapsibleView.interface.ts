export interface CollapsibleViewProps {
  question: string;
  answer: string;
}
