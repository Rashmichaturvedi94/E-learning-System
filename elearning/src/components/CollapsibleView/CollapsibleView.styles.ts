import styled from 'styled-components/native';
import { CollapsibleViewProps } from './CollapsibleView.interface';
import { Text } from 'react-native';

export const Question = styled(Text)`
  font-size: 20px;
  font-weight: bold;
  margin: 16px 16px;
`;

export const Answer = styled(Text)`
  font-size: 15px;
  font-weight: bold;
  margin: 0px 16px;
  margin-bottom: 16px;
  color: gray;
`;
