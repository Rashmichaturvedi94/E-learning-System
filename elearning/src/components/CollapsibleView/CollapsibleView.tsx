import React, { FC, useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import Collapsible from 'react-native-collapsible';
import { Question, Answer } from './CollapsibleView.styles';
import { CollapsibleViewProps } from './CollapsibleView.interface';

export const CollapsibleView: FC<CollapsibleViewProps> = ({
  question,
  answer,
}) => {
  const [isCollapsed, setIsCollapsed] = useState(true);
  return (
    <View>
      <TouchableOpacity onPress={() => setIsCollapsed((oldValue) => !oldValue)}>
        <Question>{question}</Question>
      </TouchableOpacity>
      <Collapsible collapsed={isCollapsed}>
        <Answer>{answer}</Answer>
      </Collapsible>
    </View>
  );
};
