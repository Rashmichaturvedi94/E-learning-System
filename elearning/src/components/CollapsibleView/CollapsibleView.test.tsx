import React from 'react';
import { render } from '@testing-library/react-native';
import { CollapsibleView } from './CollapsibleView';

describe('components/CollapsibleView', () => {
  it('should render', () => {
    const component = render(<CollapsibleView />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
