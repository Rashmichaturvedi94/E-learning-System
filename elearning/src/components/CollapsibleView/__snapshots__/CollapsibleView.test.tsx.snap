// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CollapsibleView should render 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 16,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 16,
          },
        ]
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      forwardedRef={[Function]}
      onLayout={[Function]}
      style={Object {}}
    >
      <Text
        style={
          Array [
            Object {
              "color": "gray",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 0,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
